composer/callbacks/checkpoint_saver.py

```diff
@@ -55,20 +55,13 @@ def checkpoint_periodically(interval: Union[str, int, Time]) -> Callable[[State,
         raise NotImplementedError(
             f'Unknown checkpointing interval: {interval.unit}. Must be TimeUnit.EPOCH or TimeUnit.BATCH.')
 
-    last_checkpoint_batch: Optional[Time] = None
-
     def checkpoint_save_interval(state: State, event: Event):
-        nonlocal last_checkpoint_batch
         elapsed_duration = state.get_elapsed_duration()
         assert elapsed_duration is not None, 'elapsed_duration is set on the BATCH_CHECKPOINT and EPOCH_CHECKPOINT'
 
+        # Always checkpoint at end of training
         if elapsed_duration >= 1.0:
-            # if doing batch-wise checkpointing, and we saved a checkpoint at the batch_checkpoint event
-            # right before the epoch_checkpoint event, do not save another checkpoint at the epoch_checkpoint
-            # event if the batch count didn't increase.
-            if state.timestamp.batch != last_checkpoint_batch:
-                last_checkpoint_batch = state.timestamp.batch
-                return True
+            return True
```

Comment:
just so i understand, this is a stylistic change, not functional, correct?

Reply:
Discussed offline, this is a functional change and not a stylistic one. See:
> Update: It looks like there were actually two (!!) bugs going on here. The previous implementation of checkpoint_save_interval was not idempotent, so when EMA was modified last week to move the EMA weights into the default weight slot on a checkpoint event, it broke checkpointing for end of training. Fixed checkpoint_save_interval to be idempotent.